### DISCUSSION TITLE
stat/xxx/POWER = ON msg to be published when light turns on as a result of eg a dimmer cmnd

### DISCUSSION THREAD

**Discussion by smcgann99** [arendst/Tasmota#20054] [Ideas]
The above topic is really useful for easily tracking changes to power state, as it is only sent when relevant. 
This makes the server side logic nice and straight forward.  However some commands eg dimmer do change this state, but do NOT publish on this topic. This seems illogical to me, as it makes the topic inconsistent, and so basically redundant.

I know that result contains the power state, but this is sent whenever any of the other values included change, not necessarily power.

**Comment by Noschvie** [comment]
Add logs for a better understanding.

**Comment by smcgann99** [comment]
Well as I described above - 

**19:06:10.298 CMD: power on**
19:06:10.308 MQT: stat/Bulb-10/RESULT = {"POWER":"ON"}
**19:06:10.313 MQT: stat/Bulb-10/POWER = ON**
**19:06:16.094 CMD: power off**
19:06:16.102 MQT: stat/Bulb-10/RESULT = {"POWER":"OFF"}
**19:06:16.106 MQT: stat/Bulb-10/POWER = OFF**
**19:06:23.454 CMD: dimmer 100** - _Doesn't generate a POWER topic_
19:06:23.467 MQT: stat/Bulb-10/RESULT = {"POWER":"ON","Dimmer":100,"Color":"0000008679","HSBColor":"0,100,0","White":100,"CT":318,"Channel":[0,0,0,53,47]}

**Reply by sfromis** [reply]
> `MQT: stat/Bulb-10/POWER = ON`

This is specifically a response to the Power command, it is not meant as a general message about power state; that is what you have in the `RESULT` topic (or command-named topic if you use `SetOption4`). To get the benefit of clean JSON, you can use `SetOption90` to disable that extra power command response, which is not adding any useful info.

**Comment by smcgann99** [comment]
The **MQT: stat/Bulb-10/POWER =** response is really useful as its only triggered by change of power state, as such it makes sense that its triggered whenever this changes, otherwise what it the point of it ?
Using RESULT means a lot of extra messages to parse to catch power state, even though many RESULT message arrive that have nothing to do with POWER state.
It wouldn't break anything to add this functionality that I can think of ?

**Reply by sfromis** [reply]
No, It is not triggered by a change of power state, but only when using the "power" command, also without a change of state. If you specifically want a signal for power state change only, you need a rule to pick this up, and publish.

You can reduce the already low amount of messages to parse using `SetOption4` to replace `RESULT` in the topic by the command name. But then you need to also check other commands able to trigger a power state change.

To be clear, I'm not arguing for or against, merely clarifying how it works now.

**Comment by smcgann99** [comment]
Thanks, I'm certainly not looking to argue either Tasmota is amazing ;-)
Just trying to understand the logic for that topic. I get it only reports if a power command was used, and not a change of power state. But I don't understand why ?

There is not similar topic published for other commands eg stat/Bulb-10/DIMMER = 100

**Reply by sfromis** [reply]
Tasmota was started many years ago, and while things have been streamlined since then, old and basically superseded features (like the one you want to expand on) tend to be retained for backwards compatibility, instead of doing breaking changes. This also means that such outdated corners are generally not included when features are added, as the non-JSON formats are not really seen as worth it to focus on. That's also why there is an option to just filter those "odd" messages away, as richer JSON works so much better with decent tools.

**Comment by smcgann99** [comment]
No problem, I will do a bit of code refactoring in Node-Red, which as you might know often expects values in payload, rather than other properties.

**Reply by sfromis** [reply]
Node-Red is great for handling the JSON payloads with multiple values
